backend/typescript/middlewares/auth.ts

```diff
@@ -87,3 +83,37 @@ export const isAuthorizedByEmail = (emailField: string) => {
     return next();
   };
 };
+
+/**
+ * isAuthorizedByUserIdFromBody verifies that the userId passed in the request
+ * body matches with the logged in user
+ *
+ * @param userIdField - the name of the userId field in the request body
+ */
+export const isAuthorizedByUserIdFromBody = (userIdField: string) => {
+  return async (
+    req: Request,
+    res: Response,
+    next: NextFunction,
+  ): Promise<Response | void> => {
+    const userId = String(req.body[userIdField]);
```

Comment:
Can we just do `const userId = req.params[userIdField] || String(req.body[userIdField]);` and just have one function?

Reply:
Might have to be careful of null access when doing something like `req.params[userIdField]` though

Reply:
Good idea! I refactored this as you suggested and the functionality remains the same.